internal/pkg/daemon/checks/post.go

```diff
@@ -198,13 +198,17 @@ func CheckBlockSignedBy(address bytes.HexBytes, height int64, consensusState jso
 	if err := json.Unmarshal(consensusState, &rs); err != nil {
 		return InvalidBlockState, errors.Wrapf(err, "Error in parsing consensus state: %v, will retry")
 	}
+
 	parts := strings.Split(rs.HeightRoundStep, "/")
 	currentHeight, err := strconv.ParseInt(parts[0], 10, 64)
 	if err != nil {
 		return InvalidBlockState, errors.Wrapf(err, "Error in parsing height from consensus state")
 	}
 
-	if currentHeight > height {
+	// when the chain halts at height N, the height returned in consensus state should be N+1
+	// because the _next_ block is being voted on
+	// so we should return BlockSkipped only if the state goes further than that
+	if currentHeight > height+1 {
```

Comment:
i think we should not change this function, rather, call `CheckBlockSignedBy(.., UPGRADE_HEIGHT+1, ..)`